pkg/kv/kvserver/kvflowcontrol/rac2/token_tracker.go

```diff
@@ -56,8 +58,8 @@ func (t *Tracker) Empty() bool {
 	// total number of tokens held, and return whether it's zero. It's also
 	// possible to make it atomic and avoid locking the mutex in replicaSendStream
 	// when calling this.
-	for pri := range t.tracked {
-		if t.tracked[pri].Length() != 0 {
+	for _, tokens := range t.deducted {
+		if tokens != 0 {
```

Comment:
Can we have zero-size deductions in `t.tracked`? If so, maybe this check should still be `Length != 0`, since it would be technically possible that zero-size entries aren't admitted yet.

Reply:
We can't have zero-sized `tokens` parameter in the call to `Track`. Feel free to add an assertion under `CrdbTestBuild`.

I think we should keep using `Length`, and add a code comment for `Empty` that states it is about the tracker state being empty (not about the tokens being zero). The existing code should be as efficient since `Length()` will inline.
<!-- Reviewable comment -OELVY_O2vycHQc0Rf9O:b-o8v3a9 -->
<!-- Sent from Reviewable.io -->